test/lib/ufe/testUIIcons.py

```diff
@@ -0,0 +1,159 @@
+            ('NodeGraph',               'out_USD_UsdTyped.png'),
+            ('NurbsCurves',             'out_USD_UsdGeomCurves.png'),
+            ('NurbsPatch',              'out_USD_NurbsPatch.png'),
+            ('OpenVDBAsset',            'out_USD_UsdGeomXformable.png'),
+            ('PackedJointAnimation',    'out_USD_SkelAnimation.png'),
+            ('PluginLight',             'out_USD_UsdGeomXformable.png' if usdVer >= (0, 21, 11) else 'out_USD_UsdLuxLight.png'),
+            ('PluginLightFilter',       'out_USD_LightFilter.png'),
+            ('PointInstancer',          'out_USD_PointInstancer.png'),
+            ('Points',                  'out_USD_Points.png'),
+            ('PortalLight',             'out_USD_UsdLuxBoundableLightBase.png' if usdVer >= (0, 21, 11) else 'out_USD_UsdLuxLight.png'),
+            ('RectLight',               'out_USD_UsdLuxBoundableLightBase.png' if usdVer >= (0, 21, 11) else 'out_USD_UsdLuxLight.png'),
+            ('RenderProduct',           'out_USD_UsdTyped.png'),
+            ('RenderSettings',          'out_USD_UsdTyped.png'),
+            ('RenderVar',               'out_USD_UsdTyped.png'),
+            ('Scope',                   'out_USD_Scope.png'),
+            ('Shader',                  'out_USD_UsdTyped.png'),
+            ('SkelAnimation',           'out_USD_SkelAnimation.png'),
+            ('SkelRoot',                'out_USD_SkelRoot.png'),
+            ('Skeleton',                'out_USD_Skeleton.png'),
+            ('SpatialAudio',            'out_USD_UsdGeomXformable.png'),
+            ('Sphere',                  'out_USD_Sphere.png'),
+            ('SphereLight',             'out_USD_UsdLuxBoundableLightBase.png' if usdVer >= (0, 21, 11) else 'out_USD_UsdLuxLight.png'),
+            ('Volume',                  'out_USD_Volume.png'),
+            ('Xform',                   'out_USD_UsdGeomXformable.png')
+        ]
+        if usdVer >= (0, 21, 11):
+            primTypes.extend([
+                ('PhysicsCollisionGroup',   'out_USD_UsdTyped.png'),
+                ('PhysicsDistanceJoint',    'out_USD_UsdTyped.png'),
+                ('PhysicsFixedJoint',       'out_USD_UsdTyped.png'),
+                ('PhysicsJoint',            'out_USD_UsdTyped.png'),
+                ('PhysicsPrismaticJoint',   'out_USD_UsdTyped.png'),
+                ('PhysicsRevoluteJoint',    'out_USD_UsdTyped.png'),
+                ('PhysicsScene',            'out_USD_UsdTyped.png'),
+                ('PhysicsSphericalJoint',   'out_USD_UsdTyped.png')
+            ])
+
+        for ty,iname in primTypes:
+            prim = self.stage.DefinePrim('/%s' % ty, ty)
+            ufeIcon = self._getUIInfoIcon(ty)
+            if ufeIcon:
+                self.assertEqual(ufeIcon, iname, msg='for prim type "%s"' % ty)
+            else:
+                iconFile = self._getAncestorNodeIcon(ty)
+                self.assertEqual(iconFile, iname, msg='for prim type "%s"' % ty)
```

Comment:
For each registered node type (including the new ones added in 21.11) create a prim of that type and then first ask the UIInfoHandler for the icon. If not found use the ancestor loop. In both cases assert that the icon found is equal to the one we expect.